### ISSUE TITLE
"topological sort failed" error occurred

### ISSUE THREAD

**Issue by abcang** [winebarrel/ridgepole#286]
"topological sort failed" error occurred when foreign key constraints are set on each other. It did not occur in v0.8.0 but now occurs in v0.8.2.

Can be reproduced in the following ways.

```rb
create_table "invites", force: :cascade do |t|
  t.bigint "user_id", null: false
  t.string "code", default: "", null: false
  t.index ["user_id"], name: "index_invites_on_user_id"
end

create_table "users", force: :cascade do |t|
  t.string "name", null: false
  t.bigint "invite_id"
end

add_foreign_key "invites", "users"
add_foreign_key "users", "invites"
```

```sh
$ ridgepole -Etest -c config/database.yml --apply
Apply `Schemafile`
[ERROR] topological sort failed: ["invites", "users"]
        /Users/abcang/.anyenv/envs/rbenv/versions/2.6.1/lib/ruby/2.6.0/tsort.rb:230:in `block in tsort_each'
```

The `user_id` in` invites` means the user who created the invitation, and the `invite_id` in` users` means the invitation passed through when the user was registered.
This error is not occurred if we move `invite_id` in ` users` to another table, but I don't want to change the schema too much.

```rb
create_table "invites", force: :cascade do |t|
  t.bigint "user_id", null: false
  t.string "code", default: "", null: false
  t.index ["user_id"], name: "index_invites_on_user_id"
end

create_table "users", force: :cascade do |t|
  t.string "name", null: false
end

create_table "invited_users", force: :cascade do |t|
  t.bigint "invite_id", null: false
  t.bigint "user_id", null: false
end

add_foreign_key "invites", "users"
add_foreign_key "invited_users", "users"
add_foreign_key "invited_users", "invites"
```

**Comment by abicky** [comment]
Thank you for your explanation. This issue is related to https://github.com/winebarrel/ridgepole/pull/284, so I'll fix it.

**Comment by abicky** [comment]
I've created a PR https://github.com/winebarrel/ridgepole/pull/287.

**Comment by abicky** [comment]
@abcang  I've released v0.8.3. Please try it.

**Comment by abcang** [comment]
Thank you! Now it works.
